Add Promscale install using package cloud and refactor install layout (#1228)
timescale/docs: promscale/installation/docker.md

# Install Promscale using a Docker image

> Docker images for the [Promscale Connector][promscale-docker-image] and
> [TimescaleDB (with the Promscale extension)][timescaledb-docker-image] are
> available on Docker Hub.

Reviewer: ```suggestion
You can use Docker to install 
[TimescaleDB with the Promscale extension][timescaledb-docker-image], 
and the [Promscale connector][promscale-docker-image]. 
The Docker images are available for download from Docker Hub.
```

If you are upgrading from the previously used [Alpine image][alpine-image]

> follow the[upgrade instructions][upgrade].

Reviewer: ```suggestion
follow the [upgrade instructions][upgrade].
```

# Install Promscale using a Docker image
You can use Docker to install 
[TimescaleDB with the Promscale extension][timescaledb-docker-image], 

> and the [Promscale connector][promscale-docker-image]. 

Reviewer 2: ```suggestion
and the [Promscale Connector][promscale-docker-image]. 
```

After you have installed Promscale, you can ingest data.
For instructions, see the [send data][send-data] section.

> ## Upgrading from the previous alpine image

Reviewer 2: We've moved that content to the ugprade guide but maybe we could link to that from here. Mostly because folks upgrading may actually land here before finding the upgrade instructions.

Author: Why will users land in install instructions for upgrade instructions? If we add the reference here we need to add it for all installation pages. So IMO we shouldn't reference upgrade-related content on installation instructions. This might confuse the people getting started with Promscale.

Reviewer 2: Ok. We can do some user testing later to see how people do upgrades. Maybe they don't even look at documentation at all in general.

I don't know that I would look for a specific upgrade guide but could be wrong. Maybe upgrading do look at the release notes and we could have a link to the upgrade guide there.

> Previously, our recommended image was located at [`timescaledev/promscale-extension`](https://hub.docker.com/r/timescaledev/promscale-extension).

Reviewer: ```suggestion
Previously, the recommended image was located at [`timescaledev/promscale-extension`](https://hub.docker.com/r/timescaledev/promscale-extension).
```

It was based on the [Alpine docker image for PostgreSQL](https://github.com/docker-library/postgres/blob/e8ebf74e50128123a8d0220b85e357ef2d73a7ec/12/alpine/Dockerfile).

> Because of [collation bugs](https://github.com/docker-library/postgres/issues/327) and other issues we have now switched to the Debian-based image above.

Reviewer: ```suggestion
Because of [collation bugs](https://github.com/docker-library/postgres/issues/327) in the previous version, you should use the Debian-based image instead.
```

Reviewer: I've edited this, but I'm not sure I like talking about things that are no longer true. Is it possible to not mention the old Alpine image, and just point to the new Debian one?

Reviewer 3: I agree!!!! never a good practice to document issues.

The previous Alpine-based image are updated and supported until
the end of 2022 but users are encouraged to migrate to the
`timescale/timescaledb-ha`. All new installations should switch to the
`timescale/timescaledb-ha`image.
<highlight type="note">
Migrating to Debian version can be a lengthy process and involves downtime.
</highlight>

> <procedure>

Reviewer: ```suggestion

<procedure>
```

1. Use `docker inspect` to determine the data volumes used by your database for the data directory.

> 1. Shutdown all Promscale Connectors.

Reviewer: ```suggestion
1. Shut down all Promscale Connectors.
```

> 1. Shutdown the original database docker image while preserving the volume mount

Reviewer: ```suggestion
1. Shut down the original database Docker image, but make sure you preserve the volume mount
```

   for the data directory. You need to mount this same directory in the new
   image.

> 1. Change the ownership of the data-directory to the postgres user and group in

Reviewer: ```suggestion
1. Change ownership of the data directory to the `postgres` user and group in
```

   the new image. For example:

   ```
   docker run -v <data_dir_volume_mount>:/var/lib/postgresql/data timescale/timescaledb-ha:pg14-latest chown -R postgres:postgres /var/lib/postgresql/data
   ```

> 1. Start the new docker container with the same volume mounts as that the

Reviewer: ```suggestion
1. Start the new Docker container with the same volume mounts that the
```

   original container used.

> 1. Connect to the new database using psql and reindex all the data that has
>    collatable data. Use the following query to reindex all the necessary indexes:

Reviewer: ```suggestion
1. Connect to the new database using psql and reindex all the collatable data. Use this query to reindex all the necessary indexes:
```

     END LOOP;
   END$$;
   ```
   This is necessary because the collation in the Alpine image is broken and so
   BTREE-based indexes remain incorrect until they are reindexed. It is
   extremely important to execute this step before ingesting new data to avoid
   data corruption. This process can take a long time depending on the indexed
   textual data in the database.  

1. Restart the Promscale Connector

> </procedure>

Reviewer: This entire procedure ☝️ needs more actual commands in it.

Reviewer 3: Hi @VineethReddy02  could you please provide the commands for all the steps in the procedure.

Author: I'm removing the upgrade section from install docs.

> If you are using Kubernetes instead of plain docker:

Reviewer: ```suggestion
If you are using Kubernetes instead of plain Docker:
```

<procedure>

> 

Reviewer: Needs a procedure title

1. Shutdown the Promscale Connector pods

> 1. Change the database pod to use the debian docker image and restart it.

Reviewer: ```suggestion
1. Change the database pod to use the Debian Docker image.
1. Restart the pod.
```

> 1. Execute jobs for the script in steps 4 and 6 above.

Reviewer: ```suggestion
1. Execute jobs for the script in steps 4 and 6 above.
```

Actually type out the steps here, don't reference others. You don't know if readers have seen the previous procedure, and you can't guarantee that steps 4 and 6 will remain the same over time.

Reviewer 3: Hi @VineethReddy02  for now I have copied the steps 4 and 6 to the Kubernetes section. However, the commands will be different in Kubernetes. Please provide the commands.
